src/ast/passes/semantic_analyser.cpp

```diff
@@ -3816,6 +3817,13 @@ void SemanticAnalyser::visit(AttachPoint &ap)
   } else if (ap.provider == "BENCH") {
     if (ap.target.empty())
       ap.addError() << "BENCH probes must have a name";
+    if (is_final_pass()) {
+      if (has_benchmark_.contains(ap.target))
+        ap.addError() << "\"" + ap.target + "\""
```

Comment:
You can make the `std::unordered_set` above a `std::map<std::string, Location>`. Then here, do something like (supposing it is called `benchmark_locs_`):

```
auto it = benchmark_locs_.find(ap.target);
if (it != benchmark_locs_.end()) {
  auto &err = ap.addError();
  err << ... (original error);
  err.addContext(it->second) << "this is the other instance";
}
benchmark_locs_.emplace(ap.target, ap.loc);
```

That will give a message that includes both locations.

Reply:
Done. Updated the PR description and commit message with example output.